File: gtg-frontend/src/app/models/meetUserVote.ts
import {voteItemTypeEnum} from '../shared/voteItemTypeEnum';
import {IModelBase} from './modelBase';
import {IMeetUser} from './meetUser';

export interface IMeetUserVote extends IModelBase{
  votableItemType: voteItemTypeEnum,
  votableItemId: string,
  rating: number,
  meetUser: IMeetUser
}

File: gtg-backend/Models/MeetUserVote.cs
using gtg_backend.Enums;

namespace gtg_backend.Models;

public class MeetUserVote : ModelBase
{
    public double? Rating { get; set; }
    public Guid VotableItemId { get; set; }
    public VotableItemType VotableItemType { get; set; }
    
    public Guid MeetUserId { get; set; }
    
    //Navigation Properties
    public MeetUser? MeetUser { get; set; }
}